src/core/namespace.c

```diff
@@ -787,6 +796,32 @@ static int prefix_where_needed(MountList *ml, const char *root_directory) {
         return 0;
 }
 
+static bool verity_has_duplicates(MountList *ml, const MountEntry *needle) {
+
+        assert(ml);
+        assert(needle);
+
+        if (needle->verity.root_hash_size == 0)
+                return false;
+
+        /* Overlayfs rejects supplying the same directory inode twice as determined by filesystem UUID and
+         * file handle in lowerdir=, even if they are mounted on different paths, as it resolves each mount
+         * to its source filesystem, so drop duplicates, and keep the last one. This only covers non-DDI
+         * verity images. */
+
+        for (const MountEntry *m = needle + 1; m < ml->mounts + ml->n_mounts; m++) {
```

Comment:
hmm, the start value is key here, you only scan the remainder of the list, and I think it would be wise to note that in a comment here, because this is not something that just scan the full list of mounts.

maybe even rename the function verity_has_later_duplicates()? so that it's clear this only scans the "later" part of the whole thing.